### PR TITLE
Extend application's capabilities by trait definition

### PR DESCRIPTION
**Pull request by leejanee**
Expand the application's capabilities in the following two ways

1. Patch the workload CR by describing the patch content in the definition
2. Create association CR for workload by describing the output object in the definition

- [x] _samples in config/samples/vela-server_

### CODE REVIEW

File: config/samples/vela-server/Demo.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# Vela Server
```
Comment (hongchaodeng) [inline config/samples/vela-server/Demo.md:1]: This shouldn't be called `vela server`. Probably `application`?

File: config/samples/vela-server/template.yaml [hunk 2]
```diff
@@ -59,18 +59,79 @@ spec:
...
+    name: "null"
   extension:
     template: |-
-      output: {
-      	apiVersion: "core.oam.dev/v1alpha2"
-      	kind:       "ManualScalerTrait"
-      	spec: {
-      		replicaCount: parameter.replicas
-      	}
+      patch: {
```
Comment (hongchaodeng) [inline config/samples/vela-server/template.yaml:65]: Could you add some doc?
I couldn't find any doc describing what patch does.

**Comment by hongchaodeng** [comment]
Why just update existing templates and CLI to use server side CUE?

**Comment by hongchaodeng** [comment]
Why not just update existing templates and CLI to use server side CUE?

File: config/samples/vela-server/Demo.md (hunk 1 above)
Comment (leejanee) [reply config/samples/vela-server/Demo.md:1]: Agree, I will edit it

**Review by wonderflow — APPROVED** [review]
LGTM，I think we can fix in the following PRs. @hongchaodeng

**Comment by hongchaodeng** [comment]
Sounds great! @wonderflow

**Pull request merged**